src/encode.jl

```diff
@@ -0,0 +1,53 @@
+
+function gif_encode(filepath::AbstractString, img::AbstractArray, numcolors::Int=256)
+    Error = Cint(0)
+    gifFile = LibGif.EGifOpenFileName(filepath,0, Ref(Error));
+    colors = []
+    
+    if (gifFile == C_NULL)
+        println("EGifOpenFileName() failed - ");
+        return false;
+    end
+
+    # generation of a colormap
+    img = quantization(img, num)
+    append!(colors, unique(img))
```

Comment:
This seems like what DitherPunk does.

Reply:
Yes, it has color quantization but it's extremely slow due to use K-means for larger colormaps like 256 colors that I am testing on.
```
julia> @btime get_colorscheme(img, 256)
  594.995 ms (13873 allocations: 10.10 MiB)
256-element Array{Lab,1} with eltype Lab:
 Lab{Float32}(0.0030800242f0,0.013884343f0,-0.0012285172f0)
 Lab{Float32}(55.85164f0,6.267978f0,42.197575f0)
 Lab{Float32}(14.842407f0,16.811573f0,21.251535f0)
 ⋮
 Lab{Float32}(37.92864f0,9.700211f0,33.112446f0)
 Lab{Float32}(42.646324f0,9.169061f0,40.786533f0)
 Lab{Float32}(22.497734f0,7.0694623f0,19.340601f0)
 ```
Method: https://github.com/JuliaImages/DitherPunk.jl/blob/master/src/clustering.jl
Resource on this topic: https://stackoverflow.com/questions/49710006/fast-color-quantization-in-opencv

Reply:
I think what @roflmaostc is doing in Noise.jl is uniform quantization, which is a very simple method too

Reply:
We might be interested in having method similiar to minimum variance quantization which is said to be much better than above two methods.

Reply:
Now we have 2 more quantizers: median cut algorithm and octree color quantisation algorithm. Now that I am doing it, I will get Kd-tree quantisation method also in next few commits but testing them properly is going to be a task.

Reply:
We have KDtree color quantisation method too now(covering base for all standard color quantiser methods) but now we need to thoroughly test them and ensure they don't break. Number of memory allocations is an issue and also benchmark time still concerning.